### PR TITLE
chore(emx2): spark to javalin migration

### PR DESCRIPTION
**Pull request by harmbrugge**
What are the main changes you did:
- Migrated the API code from Spark to Javalin

how to test:
Please test all features of emx2 to see if everything behaves as expected. Everything should work as before.

todo:
- [x] migrate api code to javalin

### CODE REVIEW

**Review by dtroelofsprins — CHANGES_REQUESTED** [review]
Did some functional tests:
Successful:
- [x] Create Directory schema from template including demo data
- [x] Schema editor 
- [x] DataCatalogueTemplate 
- [x] Fair data hub template
- [x] Data upload (zipfile with several csv's) using UI
- [ ] Other templates STILL NEED TO BE TESTED 

Not working properly:
- [ ] Create ERN Dashboard from template when going to the schema a white screen is shown, should go to /tables/#
- [ ] Directory App => footer is broken, pictures not shown (f.e. logo)
- [ ] Manually adding data => FAILS WITH ERROR: `Parsing of graphql variables failed. Should be an object with each graphql variable a key. argument "content" is null: argument "content" is null`
- [ ] Manually updating data => FAILS WITH ERROR `Parsing of graphql variables failed. Should be an object with each graphql variable a key. argument "content" is null: argument "content" is null`
- [ ] Run a script on the server => Cannot create a script due to saving error (Running the hello world script works)
- [ ] Run a script locally using the EMX2-pyclient => 
    -  create schema using template WORKS
    - downloading real BBMRI-ERIC biobanks fails with error `UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe8 in position 8688: invalid continuation byte` => using the same script, with the same data on emx2.dev.molgenis.org works fine.

**Review by dtroelofsprins — CHANGES_REQUESTED** [review]
Did some functional tests:
Successful:
- [x] Create Directory schema from template including demo data
- [x] Schema editor 
- [x] Use DataCatalogueTemplate 
- [x] Fair data hub template
- [x] Data upload (zipfile with several csv's) using UI
- [x] Other templates => tested several, but not all
- [x] Schedule a script
- [ ] Failure email when script fails => NOT POSSIBLE on preview server 

Fixed:
- [x] Directory App => footer is broken, pictures not shown (f.e. logo)
- [x] Manually adding data
- [x] Manually updating data
- [x] Create ERN Dashboard from template when going to the schema a white screen is shown, should go to /tables/#
- [x] Run a script on the server
- [x] Run a script locally using the EMX2-pyclient => 
    -  create schema using template WORKS
    -  downloading real BBMRI-ERIC biobanks

New issues (probably not tested before):
- Directory App => Load json settings does not have the proper layout
- CatalogueTemplate => SSR Catalogue shows white screen => Might also be due to recent changes to the catalogue that it's not working properly

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java [hunk 1]
```diff
@@ -46,27 +46,20 @@ public class FAIRDataPoint {
...
 
-  /**
-   * Constructor
-   *
-   * @param request
-   * @param schemas
-   * @throws Exception
-   */
-  public FAIRDataPoint(Request request, Schema... schemas) {
+  public FAIRDataPoint(Context ctx, Schema... schemas) {
```
Comment (mswertz) [inline backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java:49]: For another time: fair data point should not need ctx but instead return something that the service can render
(in other words: this molgenis-emx2-fairdatapoint package should now know about javalin(

File: backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisSessionManager.java [hunk 2]
```diff
@@ -96,46 +86,14 @@ public void clearAllCaches() {
    * Because we cannot access jetty outside spark, we override SparkJava EmbeddedServersFactory to
    * add custom session listener for session create/destroy logic
    */
-  private void createCustomJettyServerFactoryWithCustomSessionListener() {
```
Comment (mswertz) [inline backend/molgenis-emx2-webapi/src/main/java/org/molgenis/emx2/web/MolgenisSessionManager.java:99]: this is not needed anymore? That is great!

**Review by mswertz — COMMENTED** [review]
looks good.
I tested:
* catalogue app
* simple form and explorer using pet store
* set theme to use another color
* viewed a sql report

**Review by mswertz — APPROVED** [review]
looks good.
I tested:
* catalogue app
* simple form and explorer using pet store
* set theme to use another color
* viewed a sql report

**Comment by harmbrugge** [comment]
> go to https://preview-emx2-pr-4165.dev.molgenis.org/apps/ui/#/ select a schema reload page: 'File not found' error same issue for sub pages

Should be fixed now

**Comment by dtroelofsprins** [comment]
> Did some functional tests: Successful:
> 
> * [x]  Create Directory schema from template including demo data
> * [x]  Schema editor
> * [x]  Use DataCatalogueTemplate
> * [x]  Fair data hub template
> * [x]  Data upload (zipfile with several csv's) using UI
> * [x]  Other templates => tested several, but not all
> * [x]  Schedule a script
> * [ ]  Failure email when script fails => NOT POSSIBLE on preview server
> 
> Fixed:
> 
> * [x]  Directory App => footer is broken, pictures not shown (f.e. logo)
> * [x]  Manually adding data
> * [x]  Manually updating data
> * [x]  Create ERN Dashboard from template when going to the schema a white screen is shown, should go to /tables/#
> * [x]  Run a script on the server
> * [x]  Run a script locally using the EMX2-pyclient =>
>   
>   * create schema using template WORKS
>   * downloading real BBMRI-ERIC biobanks
> 
> New issues (probably not tested before):
> 
> * Directory App => Load json settings does not have the proper layout
> * CatalogueTemplate => SSR Catalogue shows white screen => Might also be due to recent changes to the catalogue that it's not working properly

From these two new issues, the first one, loading the JSON settings in the Directory App is still not working OK.
Second thing: It seems the refback fix that Morris merged last week is not included in here? As creating a schema with Directory template takes a whole lot longer than it currently does in production.

**Review by esthervanenckevort — DISMISSED** [review]
- Should update with master and resolve merge conflicts.
- There is some inconsistency in using static function call vs instance method calls, but I don't see this as a blocking issue since the code does work and part of that issue was already in the existing code base.

Directory app works
Catalogue (old app) gives error. Probably not related to this PR, but to the data model changes
SSR catalogue seems to work fine
Table explorer, editing of data and downloads in different formats work

**Pull request merged**